This patch file is from Qualcomm Innovation Center, Inc. and is provided
under the following license:

Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
SPDX-License-Identifier: BSD-3-Clause-Clear
From 6c7404a291a2a96a4ea0c9ec228772e201ab89a2 Mon Sep 17 00:00:00 2001
From: ayukuma <quic_ayukuma@quicinc.com>
Date: Tue, 10 Sep 2024 10:10:15 +0000
Subject: [PATCH] ct_status_dnat_v4_patch_02

This patch will help to configure ct status dnat rule only for ipv4.

Signed-off-by: Ayush Kumar<quic_ayukuma@quicinc.com>
---
 root/usr/share/firewall4/templates/ruleset.uc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/root/usr/share/firewall4/templates/ruleset.uc b/root/usr/share/firewall4/templates/ruleset.uc
index 7bd9309..634b5c8 100644
--- a/root/usr/share/firewall4/templates/ruleset.uc
+++ b/root/usr/share/firewall4/templates/ruleset.uc
@@ -219,7 +219,7 @@ table inet fw4 {
 		{%+ include("rule.uc", { fw4, rule }) %}
 {%  endfor %}
 {%  if (zone.dflags.dnat): %}
-		ct status dnat accept comment "!fw4: Accept port redirections"
+		meta nfproto ipv4 ct status dnat accept comment "!fw4: Accept ipv4 port redirections"
 {%  endif %}
 {%  fw4.includes('chain-append', `input_${zone.name}`) %}
 		jump {{ zone.input }}_from_{{ zone.name }}
@@ -240,7 +240,7 @@ table inet fw4 {
 		{%+ include("rule.uc", { fw4, rule }) %}
 {%  endfor %}
 {%  if (zone.dflags.dnat): %}
-		ct status dnat accept comment "!fw4: Accept port forwards"
+		meta nfproto ipv4 ct status dnat accept comment "!fw4: Accept ipv4 port forwards"
 {%  endif %}
 {%  fw4.includes('chain-append', `forward_${zone.name}`) %}
 		jump {{ zone.forward }}_to_{{ zone.name }}
-- 
2.17.1

